crates/fluvio-hub-util/src/fvm/api/download.rs

```diff
@@ -0,0 +1,186 @@
+//! Download API for downloading the artifacts from the server
+
+use std::path::{PathBuf};
+use std::io::{Cursor, copy, Read};
+use std::fs::File;
+
+use anyhow::{Error, Result};
+use http_client::async_trait;
+use sha2::{Sha256, Digest};
+use surf::StatusCode;
+
+use crate::fvm::Artifact;
+
+/// Generates the Sha256 checksum for the specified file
+fn shasum256(file: &File) -> Result<String> {
+    let meta = file.metadata()?;
+    let mut hasher = Sha256::new();
+    let mut buffer = vec![0; meta.len() as usize];
+
+    file.read_exact(&mut buffer)?;
+    hasher.update(buffer);
+
+    let output = hasher.finalize();
+    Ok(hex::encode(output))
+}
+
+/// Verifies downloaded artifact checksums against the upstream checksums
+async fn checksum(artf: &Artifact, path: &PathBuf) -> Result<()> {
```

Comment:
duplicat with other? this should be used

Reply:
Yeah! Forgot to clean up first sorry!
Now I have this centralized in utils!